chapters/annotations.tex

```diff
@@ -383,20 +383,23 @@ \section{Annotations for Simulations}\label{annotations-for-simulations}
 \end{center}
 
 \begin{annotationdefinition}[experiment]
-% henrikt-ma 2021-04: Seems strange to allow 'experiment' completely without list of options -- what would it mean?
 \begin{synopsis}[grammar]\begin{lstlisting}
-"experiment"
-   [ "(" [ experimentOption { "," experimentOption } ] ")" ]
-
-experimentOption:
-   "StartTime" "=" [ "+" | "-" ] UNSIGNED-NUMBER
-   | "StopTime" "=" [ "+" | "-" ] UNSIGNED-NUMBER
-   | "Interval" "=" UNSIGNED-NUMBER
-   | "Tolerance" "=" UNSIGNED-NUMBER
+annotation "experiment" "(" experimentModifiers ")"
 \end{lstlisting}\end{synopsis}
 \begin{semantics}
-The \lstinline{experiment} annotation defines the default start time (\lstinline!StartTime!) in {[}s{]}, the default stop time (\lstinline!StopTime!) in {[}s{]}, the suitable time resolution for the result grid (\lstinline!Interval!) in {[}s{]}, and the default relative integration tolerance (\lstinline!Tolerance!) for simulation experiments to be carried out with the model or block at hand.
+The modifiers are for the the following record.
+\begin{lstlisting}[language=modelica]
+record Experiment
+  Real StartTime(unit = "s") = 0;
+  Real StopTime(unit = "s");
+  Real Interval(unit = "s") = $\ldots{}$;
+  Real Tolerance = $\ldots{}$;
+end Experiment;
+\end{lstlisting}
+The \lstinline!experiment! annotation defines the default start time (\lstinline!StartTime!) in {[}s{]}, the default stop time (\lstinline!StopTime!) in {[}s{]}, the suitable time resolution for the result grid (\lstinline!Interval!) in {[}s{]}, and the default relative integration tolerance (\lstinline!Tolerance!) for simulation experiments to be carried out with the model or block at hand.
 If \lstinline!StartTime! is not specified it is assumed to be \lstinline!0.0!.
+The  \lstinline!StopTime! has no default and must thus be specified.
```

Comment:
I'm adding my own variant of @maltelenz's and @eshmoylova's remarks here.

Note that not providing a default value in the pseudo-code record definition is not the same as making the field required — just so that we're not afraid of leaving `StopTime` without a default in the pseudo-code definition above.  Instead, one has to explain in the text what leaving such a field undefined means.  This is what is done here, but it seems that we don't agree regarding the current formulation.

My suggestion would be to couple this to the definition of _simulation model_, with a cross reference to a proper definition in https://specification.modelica.org/master/introduction1.html#some-definitions.
```suggestion
The  \lstinline!StopTime! is required for a simulation model, see \cref{def:simulation-model}.
```

Note that the definition with `\label{def:simulation-model}` would also need to be added, but for this we need to merge #3121 first.  (Also note connection with #3146.)

Reply:
It should be the other way round, and I want to avoid depending on other PRs.

Reply:
It seems to me that finishing #3146 shouldn't take that long, so it might be worth keeping this conversation open just a little longer…

Reply:
Stating that a simulation model must have a StopTime doesn't seem necessary; the logic should be that StopTime (in a non-partial model) requires that it is a simulation model - not the other way round.

That means that
```
model A
  Real x;
equation
  der(x)=1-x;
end A;
```
remains a valid model that can be simulated, and all non-partial models with StopTime in MSL should be simulation models and it should thus be possible to simulate them.